pkg/virt-operator/install-strategy/create.go

```diff
@@ -1790,43 +1949,57 @@ func createOrUpdateValidatingWebhookConfigurations(kv *v1.KubeVirt,
 			}
 		}
 
-		injectOperatorMetadata(kv, &webhook.ObjectMeta, version, imageRegistry, id, true)
 		if !exists {
+			setLastAppliedConfiguration(webhook)
+
 			expectations.ValidationWebhook.RaiseExpectations(kvkey, 1, 0)
-			_, err := clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(webhook)
+			_, err = clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Create(webhook)
 			if err != nil {
 				expectations.ValidationWebhook.LowerExpectations(kvkey, 1, 0)
 				return fmt.Errorf("unable to create validatingwebhook %+v: %v", webhook, err)
 			}
-		} else {
-			if !objectMatchesVersion(&cachedWebhook.ObjectMeta, version, imageRegistry, id, kv.GetGeneration()) || !certsMatch {
-				// Patch if old version
-				var ops []string
 
-				// Add Labels and Annotations Patches
-				labelAnnotationPatch, err := createLabelsAndAnnotationsPatch(&webhook.ObjectMeta)
-				if err != nil {
-					return err
-				}
-				ops = append(ops, labelAnnotationPatch...)
+			continue
+		}
 
-				// Add Spec Patch
-				webhooks, err := json.Marshal(webhook.Webhooks)
-				if err != nil {
-					return err
-				}
-				ops = append(ops, fmt.Sprintf(`{ "op": "replace", "path": "/webhooks", "value": %s }`, string(webhooks)))
+		cachedWebhook.APIVersion = webhook.APIVersion
+		cachedWebhook.Kind = webhook.Kind
 
-				_, err = clientset.AdmissionregistrationV1beta1().ValidatingWebhookConfigurations().Patch(webhook.Name, types.JSONPatchType, generatePatchBytes(ops))
-				if err != nil {
-					return fmt.Errorf("unable to patch validatingwebhookconfiguration %+v: %v", webhook, err)
-				}
-				log.Log.V(2).Infof("validatingwebhoookconfiguration %v updated", webhook.GetName())
+		sync, obj, err := resourceShouldUpdate(cachedWebhook, webhook)
+		if err != nil {
+			return err
+		}
 
-			} else {
-				log.Log.V(4).Infof("validatingwebhookconfiguration %v is up-to-date", webhook.GetName())
+		if !certsMatch || sync {
```

Comment:
So if the certs are out of sync we force a reconcile? You're probably right, but I'm trying to wrap my head around that.

Reply:
correct. That keeps the current logic, so that if the kubevirt-ca secrets ca bundle changes we also update it for the webhooks.